theurgy.lic

```diff
@@ -2,185 +2,453 @@
-    else
-      EquipmentManager.new.empty_hands
-    end
+    false
+  end
+
+  def commune_ready?(commune)
+    !Flags["theurgy-#{commune[:name]}"]
+  end
+
+  def can_commune?
+    return false if @communes.empty?
+    return true unless Flags['theurgy-commune']
+    return Flags['theurgy-commune'].include?(
+             'fully prepared to seek assistance from the Immortals once again')
+  end
+  
+  def perform_commune(commune)
+    stow_hands
+    # Recheck item presence since supplies may have been used up.
+    return false if commune[:items].any? { |i| !on_hand?(i) }
+
+    send(commune[:method])
+  end
+
+  def commune_eluned
+    walk_to @data['dirt_foraging']['id']
+    forage? 'dirt'
+    
+    get_from_container(@water_holder)
+    match = bput('commune eluned',
+                 'completed this commune too recently',
+                 'You grind some dirt in your fist')
+    put_in_container(@water_holder)
+    fput('drop dirt') if [left_hand, right_hand].include?('dirt')
+
+    match == 'You grind some dirt in your fist'
   end
 
-  def comm_truff
-    fput "stow #{GameObj.right_hand.noun}" if GameObj.right_hand.name != 'Empty'
-    fput "stow #{GameObj.left_hand.noun}" if GameObj.left_hand.name != 'Empty'
+  def commune_tamsine
+    return false unless have_holy_water?
+    
+    pause 1
+    waitrt?
+    get_from_container(@water_holder)
+    fput "sprinkle #{@water_holder} on #{checkname}"
+    match = bput('commune tamsine',
+                 'completed this commune too recently',
+                 'You feel warmth spread throughout your body')
+    put_in_container(@water_holder)
+
+    match == 'You feel warmth spread throughout your body'
+  end
+
+  def commune_truffenyi
     get_from_container('Taffelberries')
-    bput('commune truffenyi', 'completed this commune too recently', 'The power of Truffenyi has answered your prayer')
+    bput('commune truffenyi',
+         'completed this commune too recently',
+         'The power of Truffenyi has answered your prayer')
     pause 1
     if right_hand.include? 'orb'
       fput 'drop orb'
+      return true
     else
       fput 'get Taffelberries'
       put_in_container('Taffelberries')
+      return false
     end
   end
-
-  ######################## SHOPPING ########################
-
-  def check_supplies
-    ensure_copper_on_hand(600, @hometown)
-    if DRStats.circle > 60
-      buy_taffelberries unless inside?('Taffelberries', @theurgy_supply_container)
+  
+  def commune_kertigen
+    if inside_container?('holy oil')
+      get_from_container('holy oil')
+    else
+      get_from_container('some oil')
+      complete_or_interrupt_research
+      cast_spell({'abbrev' => 'bless',
+                  'mana' => 1,
+                  'prep_time' => 5,
+                  'cast' => 'cast oil'}, @settings)
+      continue_research
+    end
+    
+    if right_hand.include? 'holy oil'
+      fput "sprinkle oil on #{checkname}"
```

Comment:
Should probably be a bput. We like to use those over fput's due to them being more explicit, safer, and providing more feedback.